### DISCUSSION TITLE
Trouble Parsing Station Metadata File in SourceSpec

### DISCUSSION THREAD

**Discussion by sadia145** [SeismicSource/sourcespec#52] [Q&A] [answered]
I'm new to seismic data analysis and currently working on a project using SourceSpec v1.8. I'm encountering an issue when trying to parse a station metadata file (station_metadata.xml). Despite editing the file according to the provided specifications and ensuring that the path in the configuration file is correct, I'm still getting an error indicating that the file cannot be parsed.

Here's the error message I'm receiving:

```
Unable to parse file "C:Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml" as Inventory
[Errno 2] No such file or directory: 'C:Users\\Ahsan\\Documents\\Sadia\\sourcespec-1.8\\paz\\station_metadata.xml'
```

As a newcomer to this field, I'm not entirely sure what might be causing this issue. I've checked the file path, file permissions, and the content of the station_metadata.xml file, and everything seems to be in order. 

Is there anything specific I might be missing or any other troubleshooting steps I should try to resolve this issue? Any insights or suggestions would be greatly appreciated.**

**Comment by claudiodsf** [comment]
Dear Sadia, could you please share your config file?

**Reply by sadia145** [reply]
[source_spec.txt](https://github.com/user-attachments/files/15802769/source_spec.txt)

this is config file

**Comment by krisvanneste** [comment]
I note there's a backslash missing between `C:` and `Users`. Not sure if this is because of path manipulations in sourcespec and if this is really causing the problem.

**Reply by claudiodsf** [reply]
Yes, I was thinking exactly at this 😉 and wanted to confirm it through the config file

**Reply by krisvanneste** [reply]
On my computer,
`os.path.exists(r'C:Users')`
returns True.

**Reply by sadia145** [reply]
I have tried multiple approaches and changed the path, but each attempt results in an error message stating, "Unable to parse file."

**Reply by claudiodsf** [reply]
Have you tried with a relative path?

**Reply by krisvanneste** [reply]
Could you try this in a python console:
`os.path.exists(r'C:Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml')` 
and see if it is different from:
`os.path.exists(r'C:\Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml')` ?

**Comment by sadia145** [comment]
Yes, I tried different paths, but none of them worked. In the Python shell, it indicates:
"The file at C:\Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml exists."

**Reply by claudiodsf** [reply]
So maybe the error message is not correct. Could you please also share the StationXML file?

**Reply by sadia145** [reply]
This is the full error that i am facing 
This is SourceSpec v1.8.

```
Reading station metadata...
Reading station metadata from file: C:Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml
Unable to parse file "C:Users\Ahsan\Documents\Sadia\sourcespec-1.8\paz\station_metadata.xml" as Inventory
[Errno 2] No such file or directory: 'C:Users\\Ahsan\\Documents\\Sadia\\sourcespec-1.8\\paz\\station_metadata.xml'
Reading station metadata: done
---------------------------------------------------
Event ID: 20240523_0000
Hypocenter:
  Longitude: 53.6490°, Latitude: 27.1920°, Depth: 8.6 km, Origin time: 2024-05-23T00:00:00.000000Z
Magnitude: mb 5.1
---------------------------------------------------
Reading traces...
II.UOSS.00.BH1: Unknown channel orientation: "1": skipping trace
II.UOSS.00.BH2: Unknown channel orientation: "2": skipping trace
II.UOSS.00.BHZ: cannot get instrtype from inventory: inventory is empty: skipping trace
Reading traces: done
---------------------------------------------------
No trace loaded
```

**Reply by claudiodsf** [reply, marked as answer]
Hi @sadia145,
I got your metadata file via email, but I cannot find it here, so let me put it again for reference:

[station_metadata.txt](https://github.com/user-attachments/files/15815030/station_metadata.txt)

Your file is a mix of PAZ (first 42 lines) and StationXML (next lines), which is not standard.
However, this is not the problem here.

The problem is that SourceSpec tries to read the metadata file as a PAZ file, and PAZ file reading has a bug in SourceSpec v1.8.
This is the output on macOS:

```
Reading station metadata...
Reading station metadata from file: station_metadata.txt
Unable to parse file "station_metadata.txt" as Inventory
Unable to parse file "station_metadata.txt" as PAZ file. Error at line None: I/O operation on closed file.
Reading station metadata: done
```

I got via mail a report for the same problem from another user (on Linux) three days ago, and the error is fixed in [this commit](https://github.com/SeismicSource/sourcespec/commit/b4e50351bee249b67154cb3724a34620647bd622). 

With the current development version of SourceSpec (1.8+8.gdb76727), I can read the file correctly (with limitations inherent to the PAZ format):

```
Reading station metadata...
Reading station metadata from file: station_metadata.txt
Using generic trace ID for PAZ file station_metadata.txt
Cannot find input units for ID "XX.GENERIC.XX.XXX". Defaulting to M/S
Reading station metadata: done
```

Could you please update to the current development version and try again?

Installation instructions are [in the documentation](https://sourcespec.readthedocs.io/en/latest/installation.html#installing-a-developer-snapshot).

**Reply by sadia145** [reply]
thankyou for your time.. but i couldn't find the latest version here "https://github.com/SeismicSource/sourcespec/releases" .

**Reply by claudiodsf** [reply]
There is no new release yet. You have to follow the instructions [in the documentation](https://sourcespec.readthedocs.io/en/latest/installation.html#installing-a-developer-snapshot) on how to install a developer version.

**Comment by sadia145** [comment]
I am confused because in the README.md file, you mentioned, "Edit the config file variable station_metadata to point to the paz directory," and later stated, "Note that SEED RESP and PAZ formats do not contain station coordinates, which should therefore be in the trace header (traces in SAC format)."

There are two subdirectories: "sac_data," where I placed SAC files (trace files), and "paz," where I placed poles and zeroes files in PAZ format. Do I need to point station_metadata in the config file to the "paz" folder that contains poles and zeroes, or to "sac_data," where SAC files contain station information in the headers?

**Reply by claudiodsf** [reply]
`station_metadata` should point to your `paz` directory, and trace path should be provided through the command line using the option `-t sac_data` 😉

**Comment by claudiodsf** [comment]
Hi @sadia145, could you please tell us if your problem is fixed? Thanks 😉

**Reply by sadia145** [reply]
After installing the development version as you suggested, the program is now running perfectly. Thank you very much for your help!

**Reply by claudiodsf** [reply]
Great! Thanks for the feedback!
